### PR TITLE
fix: [CDS-47301]: Adds support to show confirmation modal when template with duplicate identifier is saved

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

Issue - If new template is created with already existing identifier at a given scope(account, org, project) - it gets merged with the existing template with a new version label - this is expected behaviour, but it creates confusion sometimes to the user since he intended to create new template

Fix -> Support has been added to show a warning modal in this scenario.

#### Screenshots

https://user-images.githubusercontent.com/96409598/228208515-5cf5faf5-d5e6-4518-9e9b-fa6f10e2014e.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/hooks/useTemplateAlreadyExistsDialog.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    cancelButtonText: getString('cancel'),
+    contentText: getString('pipeline.templateAlreadyExists.content'),
+    titleText: getString('pipeline.templateAlreadyExists.title'),
+    confirmButtonText: getString('pipeline.templateAlreadyExists.confirmation'),
+    intent: DialogIntent.WARNING,
+    className: dialogClassName,
+    onCloseDialog: /* istanbul ignore next */ async isConfirmed => {
+      if (isConfirmed) {
+        try {
+          await onConfirmationCallback()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/hooks/useTemplateAlreadyExistsDialog.ts:38]: Why do we need aync...await here ?

File: src/modules/70-pipeline/hooks/useTemplateAlreadyExistsDialog.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  onConfirmationCallback: () => Promise<void>
+  onCloseCallback?: () => void
+  dialogClassName?: string
+}
+
+export interface TemplateAlreadyExistsDialogReturnType {
+  openTemplateAlreadyExistsDialog: () => void
+}
+
+export function useTemplateAlreadyExistsDialog({
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/hooks/useTemplateAlreadyExistsDialog.ts:22]: Should this be part of 72-template module?

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by satyam-harness — APPROVED** [review]

File: src/modules/70-pipeline/hooks/useTemplateAlreadyExistsDialog.ts (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/hooks/useTemplateAlreadyExistsDialog.ts]: agreed, not required, removed it

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/framework/Templates/TemplateConfigModal/TemplateConfigModal.tsx [hunk 3]
```diff
@@ -182,6 +186,19 @@ const BasicTemplateDetails = (
     [allowedScopes]
   )
 
+  const { openTemplateAlreadyExistsDialog } = useTemplateAlreadyExistsDialog({
+    onConfirmationCallback: async () => {
```
Comment (chetanvekariya-harness) [inline src/framework/Templates/TemplateConfigModal/TemplateConfigModal.tsx:190]: This is related to my previous comment. When I commented, I also expected you would remove `async` and `return Promise.resolve()` from here. I guess they are not needed.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/framework/Templates/TemplateConfigModal/TemplateConfigModal.tsx (hunk 3 above)
Comment (samarth-sharma31) [reply src/framework/Templates/TemplateConfigModal/TemplateConfigModal.tsx:190]: This is required actually - kindly check the other usage of useTemplateAlreadyExistsDialog in file -> src/modules/72-templates-library/components/TemplateStudio/SaveTemplatePopover/SaveTemplatePopover.tsx

There we need to await on template save and then do error handling.

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**